### PR TITLE
[GNNLux] Added SGConv 

### PR DESCRIPTION
**Pull request by rbSparky**
Adding conv layer according to [plan for GNNLux](https://github.com/CarloLucibello/GraphNeuralNetworks.jl/issues/461)

### CODE REVIEW

File: GNNlib/src/layers/conv.jl [hunk 1]
```diff
@@ -723,3 +723,57 @@ function d_conv(l, g::GNNGraph, x::AbstractMatrix)
     end
     return h .+ l.bias
 end
+
+####################### SGConv ######################################
+
+function sg_conv(l, g::AbstractGNNGraph, x, edge_weight::EW, conv_weight::CW) where
```
Comment (CarloLucibello) [inline GNNlib/src/layers/conv.jl:729]: the `sg_conv` function already exists, all this should be removed

File: GNNLux/test/layers/conv_tests.jl [hunk 2]
```diff
@@ -77,5 +77,10 @@
 
         #TODO test edge
     end
+
+    @testset "SGConv" begin
+        l = SGConv(in_dims => out_dims, relu)
```
Comment (CarloLucibello) [inline GNNLux/test/layers/conv_tests.jl:82]: the second argument to the constructor is not an activation function. 
I suggest running GNNLux tests locally (they are quite fast) and make sure they pass.

File: GNNLux/src/layers/conv.jl [hunk 3]
```diff
@@ -515,4 +515,60 @@ function Base.show(io::IO, l::GATv2Conv)
...
+end
+
+function SGConv(ch::Pair{Int, Int}, k = 1;
+                init_weight = glorot_uniform,
+                init_bias = zeros32,
+                use_bias::Bool = true,
+                add_self_loops::Bool = true,
+                use_edge_weight::Bool = false)
+    in_dims, out_dims = ch
+    return SGConv(in_dims, out_dims, use_bias, add_self_loops, use_edge_weight, init_weight, init_bias, k)
```
Comment (CarloLucibello) [inline GNNLux/src/layers/conv.jl:538]: the arguments passed to the constructor are not in the correct order. Please be more careful before committing and run tests locally. I cannot go over every tiny detail.

Comment (rbSparky) [reply GNNLux/src/layers/conv.jl:538]: will check

File: GNNLux/test/layers/conv_tests.jl (hunk 2 above)
Comment (rbSparky) [reply GNNLux/test/layers/conv_tests.jl:82]: will do

**Comment by rbSparky** [comment]
Fixing tests

File: GNNLux/src/layers/conv.jl [hunk 4]
```diff
@@ -515,4 +515,60 @@ function Base.show(io::IO, l::GATv2Conv)
...
+    l.use_bias || print(io, ", use_bias=false")
+    l.add_self_loops || print(io, ", add_self_loops=false")
+    !l.use_edge_weight || print(io, ", use_edge_weight=true")
+    print(io, ")")
+end
+
+(l::SGConv)(g, x, ps, st; conv_weight=nothing, edge_weight=nothing) = 
+    l(g, x, edge_weight, ps, st; conv_weight)
+
+function (l::SGConv)(g, x, edge_weight, ps, st;
```
Comment (CarloLucibello) [inline GNNLux/src/layers/conv.jl:567]: this actually should be
```julia
(l::SGConv)(g, x, ps, st) = l(g, x, nothing, ps, st)

function (l::SGConv)(g, x, edge_weight, ps, st)
```
Fixed it in #471

**Pull request merged**